src/Compilers/CSharp/Portable/Symbols/Symbol.cs

```diff
@@ -600,6 +600,17 @@ public override bool Equals(object obj)
         }
 
         public bool Equals(ISymbol other)
+        {
+            return this.Equals(other, SymbolEqualityComparer.Default.CompareKind);
+        }
+
+        public bool Equals(ISymbol other, SymbolEqualityComparer equalityComparer)
+        {
+            return equalityComparer.Equals(this, other);
+        }
+
+        // By default we don't consider the compareKind. This can be overridden.
+        public virtual bool Equals(ISymbol other, TypeCompareKind compareKind)
```

Comment:
>ISymbol [](start = 35, length = 7)

`Symbol`?

Reply:
That comes from ISymbolInternal. We have to implement for both VB and C#, so needs to be ISymbol.

---
In reply to: [305547615](https://github.com/dotnet/roslyn/pull/37247#discussion_r305547615) [](ancestors = 305547615)

Reply:
Ideally, internal callers should not be using `ISymbol`. Can we achieve that with something like:

```C#
[Obsolete]
public bool Equals(ISymbol other)
{
    return this.Equals(other as Symbol, TypeCompareKind.AllNullableIgnoreOptions);
}

bool ISymbolInternal.Equals(ISymbol other, TypeCompareKind compareKind)
{
    return this.Equals(other as Symbol, compareKind);
}

public virtual bool Equals(Symbol other, TypeCompareKind compareKind)
{
    ...
}
```

If we need an internal method with `SymbolEqualityComparer`, perhaps:
```C#
public bool Equals(Symbol other, SymbolEqualityComparer comparer)
{
    return this.Equals(other, comparer.CompareKind);
}
```

---
In reply to: [305548108](https://github.com/dotnet/roslyn/pull/37247#discussion_r305548108) [](ancestors = 305548108,305547615)